### DISCUSSION TITLE
Getting error "Unable to populate missions route"

### DISCUSSION THREAD

**Discussion by GPinks** [EDCD/EDDI#2335] [EDDI Standalone Q&A] [answered]
I am getting the above error being spoken when I am in mission monitor and I hit the Get route button.   I have had this problem for a while.   I do like the mission monitor but it is not working.
Thanks for any help.
Glen Pinkston

**Comment by Tkael** [comment]
I'm trying to find an error message matching your description. The closest that I can find is "Unable to calculate missions route." This could be spoken if you try to calculate the mission route while the current star system has not yet been defined.

**Comment by GPinks** [comment]
Greetings,   

It is what you said   “Unable to calculate missions route”.   I am sending a verbose log.   Now a couple of things.   I did have Voice attack active but had turned it off.   I do not know if this makes a difference.   At this time of the log,  I had two missions in the log but when I hit the “Get route” button, It spoke the error “Unable to calculate mission route”

**Comment by Tkael** [comment]
Do the missions shown in the mission monitor include destination system names? Odyssey missions are often lacking those details after those missions have been accepted (since Odyssey mission journal events omit them). If so then EDDI would be unable to resolve any destinations for those missions.

P.S. If you attached a verbose log to your response, it was not included in the Github discussion. You would need to post it directly to the discussion thread.

**Comment by GPinks** [comment]
[eddi_issue.zip](https://github.com/EDCD/EDDI/files/8311992/eddi_issue.zip)
Here is the log.   Sorry about that.   I had replied via outlook.  I did this reply on github

**Reply by Tkael** [reply]
I'm not seeing any immediate red flags. Speculating, if this was a temporary issue then the EDSM API may have been unavailable / temporarily down and this could have prevented EDDI from fetching system coordinates?

Location StarSystem: Orishis

Missions
```json
{
  "missions": [
    {
      "targetTypeEDName": null,
      "missionid": 855114407,
      "name": "Mission_Courier_Democracy",
      "localisedname": "Internal report delivery",
      "statusEDName": "Active",
      "originsystem": "Orishis",
      "originstation": "Piccard Dock",
      "originreturn": false,
      "faction": "Workers of Orishis Independents",
      "factionstate": "None",
      "influence": "+",
      "reputation": "+",
      "chained": false,
      "onfoot": false,
      "communal": false,
      "legal": true,
      "shared": false,
      "wing": false,
      "reward": 22945,
      "commodity": null,
      "amount": 0,
      "destinationsystem": "Alkaid",
      "destinationstation": "Maxwell City",
      "destinationsystems": [],
      "communalPercentileBand": 0,
      "communalTier": 0,
      "passengertypeEDName": null,
      "passengerwanted": null,
      "passengervips": null,
      "target": null,
      "targetfaction": "Seurinohina Natural Holdings",
      "expiry": "2022-03-21T03:22:18Z"
    },
    {
      "targetTypeEDName": null,
      "missionid": 855114677,
      "name": "Mission_Courier_Democracy",
      "localisedname": "Internal report delivery",
      "statusEDName": "Active",
      "originsystem": "Orishis",
      "originstation": "Piccard Dock",
      "originreturn": false,
      "faction": "Workers of Orishis Independents",
      "factionstate": "None",
      "influence": "+",
      "reputation": "+",
      "chained": false,
      "onfoot": false,
      "communal": false,
      "legal": true,
      "shared": false,
      "wing": false,
      "reward": 23136,
      "commodity": null,
      "amount": 0,
      "destinationsystem": "Namaka",
      "destinationstation": "Cantor Platform",
      "destinationsystems": [],
      "communalPercentileBand": 0,
      "communalTier": 0,
      "passengertypeEDName": null,
      "passengerwanted": null,
      "passengervips": null,
      "target": null,
      "targetfaction": "Hooriayan Democrats",
      "expiry": "2022-03-21T03:22:18Z"
    }
  ],
  "updatedat": "2022-03-20T03:24:38Z",
  "goalsCount": 0,
  "missionsCount": 2,
  "missionWarning": 60,
  "missionsRouteList": null,
  "missionsRouteDistance": 0.0
}
```

**Comment by Tkael** [comment, marked as answer]
Investigated and closed.
